include/robot_interfaces/robot_backend.hpp

```diff
@@ -177,11 +201,29 @@ class RobotBackend
     {
         real_time_tools::set_cpu_dma_latency(0);
 
+        const double start_time =
+            real_time_tools::Timer::get_current_time_sec();
+
         // wait until first desired_action was received
         // ----------------------------
         while (!has_shutdown_request() &&
                !robot_data_->desired_action->wait_for_timeindex(0, 0.1))
         {
+            const double now = real_time_tools::Timer::get_current_time_sec();
+            if (now - start_time > first_action_timeout_)
+            {
+                Status status;
+                status.error_status = Status::ErrorStatus::BACKEND_ERROR;
+                status.error_message = "First action was not provided in time";
+
+                robot_data_->status->append(status);
+
+                std::cerr << "Error: " << status.error_message
+                          << "\nRobot is shut down." << std::endl;
+
+                this->request_shutdown();
```

Comment:
Previously, when the method was called `shutdown`, I thought it's good to be very explicit to avoid confusion with `robot_driver_->shutdown()`.  Now that it is renamed, it does not really make sense anymore, I'm going to remove it.